bin/create-test-release.sh

```diff
@@ -0,0 +1,272 @@
+			exit 1
+        fi
+
+		http_code=$(curl -sL --header "Authorization: token $GITHUB_TOKEN" --header 'Accept: application/octet-stream' $download_url --output .cache/woorelease.zip --write-out "%{http_code}")
+		if [[ ${http_code} -lt 200 || ${http_code} -gt 299 ]]; then
+            echo "Unable to download Woorelease zip package (status_code: $http_code)"
+			echo 'Download it from: https://github.com/woocommerce/woorelease/releases/latest, and extract it into the .cache/ folder manually'
+			exit 1
+		fi
+
+		unzip .cache/woorelease.zip -d .cache >/dev/null
+		rm .cache/woorelease.zip
+        echo "Woorealease successfully downloaded to '.cache/woorelease.phar'."
+	fi
+
+	# Check for Woorelease config file
+	if [[ ! -f ~/.woorelease/config ]]; then
+	 	echo 'Woorelease config file not found (~/.woorelease/config).'
+		echo 'Please follow the configuration steps here: https://github.com/woocommerce/woorelease#prerequisites-for-configuration'
+	 	exit 1
+	fi
+}
+
+check_github_status() {
+	echo ''
+	echo 'Checking GitHub project status...'
+
+	# Check if the branch 'release/VERSION' already exists, and if not, create it using 'develop' as base
+	git ls-remote --exit-code --heads origin release/$VERSION >/dev/null || {
+		CREATE_BRANCH=true
+	}
+
+	# If tag VERSION-test exists, calculate the next test suffix to use (ex: 5.3.0-test3)
+	set +e
+    git ls-remote --exit-code --tags origin "refs/tags/$VERSION-test" >/dev/null
+    if [ $? -eq 0 ]; then
+        next=2
+	    while true; do
+		    git ls-remote --exit-code --tags origin "refs/tags/$VERSION-test$next" >/dev/null || break
+		    next=$((next+1))
+	    done
+		SUFFIX="${SUFFIX}${next}"
+	fi
+    set -e
+}
+
+print_summary() {
+	echo ''
+	echo 'This script will perform the following actions:'
+	$CREATE_BRANCH && echo "  - Create a new branch named 'release/$VERSION' using 'develop' as base, and push it to GitHub"
+	echo "  - Run Woorelease simulation locally with version=$VERSION (using the 'release/$VERSION' branch)"
+	echo "  - Create a new tag named '$VERSION-$SUFFIX' in GitHub"
+	echo "  - Create a new release named '$VERSION-$SUFFIX' in GitHub (marked as 'pre-release')"
+	echo "  - Upload the generated plugin zip file ($GITHUB_PROJECT.zip) to the newly created release in GitHub"
+}
+
+create_git_branch() {
+	echo ''
+	echo "This is the first test release for $VERSION, creating 'release/$VERSION' branch..."
+	echo ''
+	echo "> git checkout develop && git pull"
+	git checkout develop && git pull
+
+	echo ''
+    echo "> git checkout -b release/$VERSION"
+	git checkout -b release/$VERSION
+
+    echo ''
+    echo "> sed -i '' \"s/^= 5.x.x - \(.*\)$/= $VERSION - \1/\" readme.txt"
+    sed -i '' "s/^= 5.x.x - \(.*\)$/= $VERSION - \1/" readme.txt
+    echo "> sed -i '' \"s/^= 5.x.x - \(.*\)$/= $VERSION - \1/\" changelog.txt"
+    sed -i '' "s/^= 5.x.x - \(.*\)$/= $VERSION - \1/" changelog.txt
```

Comment:
These `sed` commands will not work when we move to newer major versions, so I think we need to change them to use `\d+` in place of `5` or something along those lines 🤔 

This is also very brittle. For example, `woorelease` fails if the `5.x.x` changelog entry is missing from either `changelog.txt` or `readme.txt`.

I'm not sure what the best way to fix that is 🤔 
I guess the release lead can make sure the changelog entries are correct in `develop` before running the script? That sort of defeats the point of it though...

Maybe we include a step before running `woorelease` where the script stops and instructs you to "make sure changelog entries are right" or something along those lines. Once you hit `<Enter>` the script continues?

We could also split this into 2 steps; 1 for preparing the branch automatically if possible, and another for creating the release once the branch has been prepared properly?

I'm not sure what the best way is to address this 🤔

Reply:
Added logic to check for local and remote branches already created:
- if a remote branch exists... process starts with woorelease from that branch
- if a remote branch does NOT exist, but a local does... instructions are shown to check it, push it to github and re-run the script
-  if a neither a remote nor local branch exist... a new branch based on develop will be created, and an attempt to update the version number in both readme.txt and changelog.txt will be made, after that,  both files will be checked for the tags Woorelease expect (`= 5.4.0 - 2021-xx-xx =`), if they are not correct instructions are shown to check it, push it to github and re-run the script

```
Checking GitHub project status...

A local branch named release/5.6.0 already exists, but has not been pushed to the remote repository.

Make sure that both readme.txt and changelog.txt have the correct entries for the release version:
  = 5.6.0 - 2021-xx-xx =
Push the branch to the remote repository, and re-run this script.

Aborted.
```

Also, added an explicit `Aborted` message when exiting with an error, and the message `Partially completed, additional manual steps required` is shown when the release needs to be manually created in GitHub and the zip file uploaded:
```
You need to manually create a release in GitHub:
  - Go to: https://github.com/woocommerce/woocommerce-gateway-stripe/releases/new
  - Fill the form with the following values:
    Tag version   = 5.6.0-test
    Target        = release/5.6.0
    Release title = Version 5.6.0-test. Not for production.
    Description   = This version is for internal testing. It should NOT be used for production.
  - Attach the plugin zip file 'woocommerce-gateway-stripe.php'
  - Select the checkbox 'This is a pre-release'
  - Publish the release

Partially completed, additional manual steps required.
```